### PR TITLE
priorityClassName support

### PR DESCRIPTION
**Pull request by jasonhancock**
Issue: https://github.com/rancher/backup-restore-operator/issues/114
Adds the ability to specify a priority class name.

Edit 1: add the link to the issue

### CODE REVIEW

**Comment by jiaqiluo** [comment]
@jasonhancock  Can you elaborate on the use case of the  `priorityClassName` filed here?

**Comment by jasonhancock** [comment]
@jiaqiluo we make use of [priority classes](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/) in our k8s clusters to help facilitate workloads to be scheduled on our clusters and preempt other workloads. We'd like the ability to set a priority class on the pods.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: charts/rancher-backup/templates/deployment.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ spec:
         checksum/pvc: {{ include (print $.Template.BasePath "/pvc.yaml") . | sha256sum }}
     spec:
       serviceAccountName: {{ include "backupRestore.serviceAccountName" . }}
+      {{- if ne .Values.priorityClassName "" }}
```
[the remark this answers is not in the document]
Comment (jasonhancock) [reply charts/rancher-backup/templates/deployment.yaml]: updated per suggestion

**Review by jiaqiluo — APPROVED** [review]
LGTM.

**Review by ryansann — APPROVED** [review]

**Comment by jiaqiluo** [comment]
@jasonhancock Thank you for you contribution to the project!

**Pull request merged**